server/service/src/invoice/outbound_shipment/update/generate.rs

```diff
@@ -202,6 +225,29 @@ fn generate_batches_total_number_of_packs_update(
     Ok(result)
 }
 
+fn generate_tax_update_for_lines(
+    connection: &StorageConnection,
+    invoice_id: &str,
+    tax: Option<f64>,
+) -> Result<Vec<InvoiceLineRow>, UpdateOutboundShipmentError> {
+    let invoice_lines = InvoiceLineRepository::new(connection).query_by_filter(
+        InvoiceLineFilter::new()
+            .invoice_id(EqualFilter::equal_to(invoice_id))
+            .r#type(InvoiceLineRowType::StockOut.equal_to()),
+    )?;
+
+    let mut result = Vec::new();
+    for invoice_line in invoice_lines {
+        let mut invoice_line_row = invoice_line.invoice_line_row;
+        invoice_line_row.tax = tax;
+        invoice_line_row.total_after_tax =
+            calculate_total_after_tax(invoice_line_row.total_before_tax, tax);
+        result.push(invoice_line_row);
```

Comment:
Alternative would be to not update the lines but have a different logic to fetch the line tax, i.e.

if line tax == None -> use invoice tax